yarn-project/pxe/src/contract_function_simulator/execution_data_provider.ts

```diff
@@ -259,9 +259,9 @@ export interface ExecutionDataProvider {
   ): Promise<void>;
 
   /**
-   * Removes all of a contract's notes that have been nullified from the note database.
+   * Synchronizes a contract's notes by checking for and removing nullified notes.
```

Comment:
```suggestion
   * Looks for nullifiers of active contract notes and marks them as nullified in the db if a nullifier is found.
```
This doesn't synchronize notes (it doesn't load new notes), it just loads new nullifiers.